File: part1-exercises-for-rust/hardways/leetcode-maximum-subarray/solution.c
int maxSubArray(int* nums, int numsSize){
    int subsum = 0, maxsum = -2147483648;
    
    for(int i = 0; i < numsSize; i ++)
    {
        subsum += nums[i];
        if(subsum > maxsum)
        {
            maxsum = subsum;
        }
        if(subsum < 0) subsum = 0;
    }
    return maxsum;
}

File: part1-exercises-for-rust/hardways/leetcode-maximum-subarray/solution.rs
impl Solution {
    pub fn max_sub_array(nums: Vec<i32>) -> i32 {
        let mut sum = 0;
        let mut max = 0;
        let mut maxi = nums[0];
        for i in nums{
            if sum + i < 0{
                sum = 0;
            }else{
                sum = sum + i;
                if sum > max  {
                    max = sum;
                }
            }
            if i > maxi {
                maxi =i;
            }
        }
        if maxi < 0{
            max = maxi;
        }
        max
    }
}